src/shadowbox/infrastructure/json_config.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright 2018 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as fs from 'fs';
+
+import * as file_read from './file_read';
+import * as logging from './logging';
+
+export interface JsonConfig<T> {
+  // Returns the config data as a json object.
```

Comment:
I think a better comment would emphasise that this returns the actual underlying object (a public field, essentially), e.g.:
```ts
// Returns a reference (*not* a copy) to the stored data.
```